include/aie/Dialect/AIE/IR/AIE.td

```diff
@@ -1372,6 +1372,49 @@ def AIE_ObjectFifoCreateOp: AIE_Op<"objectFifo.createObjectFifo", []> {
   }];
 }
 
+def AIE_ObjectFifoLinkOp: AIE_Op<"objectFifo.link", []> {
+  let summary = "Links two objectFifos through an intermediary tile's dma";
+  let description = [{
+    The "aie.objectFifo.link" operation allows to mark two objectFifos as linked. This implies that the two objectFifos form
+    one dataflow movement which is split accross multiple objectFifos. Specifically, during the objectFifo lowering there will
+    be less memory elements generated at the link point as the two objectFifos can share.
+
+    The two objectFifos which are linked must have a link point (i.e., a shared AIE tile).
+    In L1, only objectFifos of same size may be linked. In L2, different sized objectFifos can be linked.
+
+    Example:
+    ```
+      %of_t70_t72 = AIE.objectFifo.createObjectFifo(%t70, {%t72}, 2) {sym_name = "of0"} : !AIE.objectFifo<memref<64xi16>>
+      %of_t72_t74 = AIE.objectFifo.createObjectFifo(%t72, {%t74}, 2) {sym_name = "of1"} : !AIE.objectFifo<memref<64xi16>>
```

Comment:
Would it make sense to simplify the syntax?
```suggestion
      %of_t70_t72 = AIE.objectFifo.create(%t70, {%t72}, 2) {sym_name = "of0"} : !AIE.objectFifo<memref<64xi16>>
      %of_t72_t74 = AIE.objectFifo.create(%t72, {%t74}, 2) {sym_name = "of1"} : !AIE.objectFifo<memref<64xi16>>
```

Reply:
I think it does! I will take note of this change and add it to another PR as there will be quite a lot of tests to modify and this PR needs to be ready asap! Thank you for the suggestion!